repos/system_upgrade/el7toel8/actors/cupscheck/libraries/library.py

```diff
@@ -0,0 +1,204 @@
+from leapp.libraries.stdlib import api
+from leapp.models import CupsChangedFeatures
+from leapp import reporting
+
+
+def _get_input_model(model):
+    """
+    Gets data model from an actor.
+
+    :param obj model: object of model which data will be consumed
+    """
+    return next(api.consume(model), None)
+
+
+def check_interface_scripts(facts, report_func):
+    """
+    Checks if the data model tells interface scripts are used
+    and produces a report.
+
+    :param obj facts: model object containing info about CUPS configuration
+    :param func report_func: creates report
+    """
+    title = ('CUPS no longer supports usage of interface scripts')
+    summary = ('Interface scripts are no longer supported due '
+               'security issues - an attacker could provide '
+               'malicious script which will be run during printing')
+    hint = ('Please install the queue with PPD file for the printer '
+            'if available or install the queue with generic PPD, '
+            'add *cupsFilter2 directive into PPD of installed '
+            'queue (in /etc/cups/ppd) and reinstall the queue with modified PPD. '
+            'The interface script needs to have permissions 750 and '
+            'ownership root:lp. How to write *cupsFilter2 directive '
+            'is described at https://www.cups.org/doc/spec-ppd.html#cupsFilter2 '
+            'and the script needs to be put into /usr/lib/cups/filter '
+            'or you need to use an absolute path to the script '
+            'in *cupsFilter2 directive')
+    if facts.interface:
+        args = [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Tags([reporting.Tags.DRIVERS]),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Remediation(hint=hint)
+        ]
```

Comment:
note for me: keep separate

Reply:
@pirat89 my guess - the idea about 'groups' could be a aggregation of reports based on severity:

0) 'Drivers' report will stay separate
1) all other 'HIGH' reports will be merged in 1 func
2) all 'MEDIUM' reports will be merged together in 1 func

Let me know after you have time for looking into this.